core/src/main/scala/kafka/server/KafkaBroker.scala

```diff
@@ -300,7 +300,7 @@ abstract class KafkaBroker(val config: KafkaConfig,
         logDirFailureChannel = new LogDirFailureChannel(config.logDirs.size)
 
         /* start log manager */
-        logManager = LogManager(config, initialOfflineDirs, zkClient, brokerState, kafkaScheduler, time, brokerTopicStats, logDirFailureChannel)
+        logManager = createLogManager(initialOfflineDirs)
```

Comment:
I can see why you want to do it this way, but I think the Kip500Broker and LegacyBroker classes should go their separate ways.  Duplicate most of the code and only put the really tricky stuff into a common class.  I just don't want the code riddled with optional this and maybe that.  It should be clear what the control flow is, and I don't think it can be with legacy and kip500 in the same class.